components/sidebar/sidebar_service.cc

```diff
@@ -340,6 +345,34 @@ void SidebarService::MoveItem(int from, int to) {
   UpdateSidebarItemsToPrefStore();
 }
 
+void SidebarService::UpdateItem(const GURL& old_url,
+                                const GURL& new_url,
+                                const std::u16string& old_title,
+                                const std::u16string& new_title) {
+  // Check any existing items doen't use |new_url| if |old_url| and |new_url|
+  // is different. If both are same, only title will be updated.
+  // Sidebar can't have two items with same url.
+  if (old_url != new_url &&
+      base::ranges::any_of(items_, [&new_url](const auto& item) {
+        return new_url == item.url;
+      })) {
+    return;
+  }
+
+  auto item_iter = base::ranges::find_if(
+      items_, [&old_url](const auto& item) { return (item.url == old_url); });
+  if (item_iter != items_.end()) {
+    const int index = std::distance(items_.begin(), item_iter);
+    DCHECK(IsEditableItemAt(index));
+    item_iter->url = new_url;
+    item_iter->title = new_title;
+    for (Observer& obs : observers_)
```

Comment:
nit : I believe wrapping for loop body with `{ }` complies with our convention, as the body is more than 1 physical line.